rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -1248,6 +1258,35 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
         }
     }
 
+    private String getOutputLimit(boolean logGlobalCongif, ScheduledExecution scheduledExecution, boolean limitFramework) {
+        String outputLimit = ""
+        if (logGlobalCongif) {
+            outputLimit = frameworkService.getProjectLogOutputLimit()
+            limitFramework = true
+        } else {
+            outputLimit = scheduledExecution?.logOutputThreshold
+            if (!outputLimit) {
+                outputLimit = frameworkService.getProjectLogOutputLimit()
+                limitFramework = true
+            }
+        }
+        log.info("Logging output limit: ${outputLimit}")
+        return outputLimit
+    }
```

Comment:
This logic appears to be: if enabled, use only the global limit. if not enabled, and no job limit is enabled, use the global limit. This allows the job limit to exceed the global limit, or enforces a global limit.

I think it should be changed to:

1. if no global limit is set, apply any job limit that is set.
2. if a global limit is set, and a job limit is set, apply `min(global,job)` to the determine the limit. The global limit cannot be exceeded by the job limit if it is set.
3. if only a global limit is configured, use it.

Reply:
Done